File: Ruby/Problem 2.rb
# Sum of all EVEN Fibonacci numbers less than 4 million
x = 1
y = 2
z = 0
sum = 0

while(x < 4000000)
    z = x + y
    x = y
    y = z
    if(x % 2 == 0)
        sum = sum + x
    end
end

puts sum

File: JS/Problem 2.js
// Sum of all EVEN Fibonacci numbers less than 4 million

var x = 1;
var y = 2;
var z = 0;
var sum = 0;

while(x < 4000000) {
    
    if(x % 2 === 0) {
        sum += x;
    }
    z = x + y;
    x = y;
    y = z;
}
console.log(sum);